### PR TITLE
fix(authentication-service): submission of encrypted password in authentication service.

### LINKED ISSUE #1592: @authentication-service:Submission of Passwords In Clear Text
**Describe the bug**
Few API's like  auth/login ,auth/login-token etc  from UI and facades are accepting passwords in clear text .

**Expected behavior**
1.Use proper encryption to encrypt the passwords.
2.Submission of passwords in clear on HTTP and HTTPS connection should be restricted.

**Screenshots**

**Additional context**
https://sourcefuse.atlassian.net/browse/RPMS-29261

### PR DESCRIPTION
**Pull request by Surbhi-sharma1**
submission of encrypted password in authentication service api

GH-1592

## Description

Fixes #1592 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] Performed a self-review of my own code
- [x] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: services/authentication-service/src/modules/auth/login.controller.ts [hunk 1]
```diff
@@ -370,16 +370,39 @@ export class LoginController {
...
+        );
+        const decryptedNewPassword = await this.userRepo.decryptPassword(
+          req.password,
+        );
+        newPassword = decryptedNewPassword;
+        oldPassword = decryptedOldPassword;
+      } else {
+        newPassword = req.password;
+        oldPassword = req.oldPassword;
+      }
```
Comment (yeshamavani) [inline services/authentication-service/src/modules/auth/login.controller.ts:388]: assign default value to old/newpassword from the request body and if the decryption key is present override it with the decrypted once

File: services/authentication-service/src/modules/auth/login.controller.ts [hunk 2]
```diff
@@ -370,16 +370,39 @@ export class LoginController {
...
       );
     } else {
+      if (process.env.PRIVATE_DECRYPTION_KEY) {
+        const decryptedNewPassword = await this.userRepo.decryptPassword(
+          req.password,
+        );
+        newPassword = decryptedNewPassword;
+      } else {
+        newPassword = req.password;
+      }
```
Comment (yeshamavani) [inline services/authentication-service/src/modules/auth/login.controller.ts:402]: same

File: services/authentication-service/src/repositories/user.repository.ts [hunk 3]
```diff
@@ -126,6 +127,12 @@ export class UserRepository extends DefaultUserModifyCrudRepository<
   }
 
   async verifyPassword(username: string, password: string): Promise<User> {
+    let Password: string;
```
Comment (yeshamavani) [inline services/authentication-service/src/repositories/user.repository.ts:130]: camel case

**Review by yeshamavani — CHANGES_REQUESTED** [review]

File: services/authentication-service/src/modules/auth/login.controller.ts (hunk 1 above)
Comment (Surbhi-sharma1) [reply services/authentication-service/src/modules/auth/login.controller.ts]: Done

File: services/authentication-service/src/modules/auth/login.controller.ts (hunk 2 above)
Comment (Surbhi-sharma1) [reply services/authentication-service/src/modules/auth/login.controller.ts]: Done

File: services/authentication-service/src/repositories/user.repository.ts (hunk 3 above)
Comment (Surbhi-sharma1) [reply services/authentication-service/src/repositories/user.repository.ts]: Done

**Review by yeshamavani — CHANGES_REQUESTED** [review]
please implement it through a provider

**Review by shubhamp-sf — APPROVED** [review]

**Comment by yeshamavani** [comment]
@Surbhi-sharma1  code smell

**Pull request merged**